src/type.cpp

```diff
@@ -2727,6 +2727,22 @@ const Type *FunctionType::GetAsConstType() const { return this; }
 
 const Type *FunctionType::GetAsNonConstType() const { return this; }
 
+const Type *FunctionType::GetAsUnmaskedType() const {
```

Comment:
Two things here.

1. Check if the function is already masked/unmasked, possibly no need to create a new one.
2. It's better cache the result (i.e. add the fields `asUnmaskedType`, `asMaskedType`) and ensure that `ftype-> GetAsUnmaskedType()->GetAsMaskedType() == ftype`.

It's not necessary with current usage strictly speaking, but it could be assumed, as this is how `GetAsUniformType()`/`GetAsVaryingType()` are implemented.